### PR TITLE
Userdao

### PR DESCRIPTION
**Pull request by VolodymyrTistechok**
class UserDaoImpl

### CODE REVIEW

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -4,37 +4,44 @@
...
-        return null;
+       return userMap.values().stream()
+                .filter(user -> user.getUserRole().equals(userRole))
+                .collect(toMap(User::getUsername, User -> User));
     }
 
     @Override
     public Map<String, User> getAllUsers() {
-        return null;
+        return userMap;
```
Comment (tupchiienko) [inline src/main/java/dao/impl/UserDaoImpl.java]: It must return copy of `userMap`

**Review by Oleksandr0404 — APPROVED** [review]

**Review by Oleksandr0404 — APPROVED** [review]

**Review by Oleksandr0404 — APPROVED** [review]

**Review by Oleksandr0404 — APPROVED** [review]

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -4,37 +4,44 @@
...
 import model.UserRole;
 
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 
+import static java.util.stream.Collectors.toMap;
+
 public class UserDaoImpl implements UserDao {
+    private Map<String, User> userMap = new HashMap<>();
```
Comment (tupchiienko) [inline src/main/java/dao/impl/UserDaoImpl.java]: Initialize `userMap` in constructor `UserDaoImpl()` as `new TreeMap()`

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -4,37 +4,44 @@
...
     @Override
     public Optional<User> getByUsername(String username) {
-        return Optional.empty();
+        return Optional.ofNullable(userMap.get(username));
     }
 
     @Override
     public Optional<User> update(String username, User newUser) {
-        return Optional.empty();
+          return Optional.ofNullable(  userMap.replace(username, newUser));
```
Comment (tupchiienko) [inline src/main/java/dao/impl/UserDaoImpl.java]: Remove reduntant spaces

File: src/main/java/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -4,37 +4,44 @@
...
     }
 
     @Override
     public Optional<User> getByUsername(String username) {
-        return Optional.empty();
+        return Optional.ofNullable(userMap.get(username));
     }
 
     @Override
     public Optional<User> update(String username, User newUser) {
```
Comment (tupchiienko) [inline src/main/java/dao/impl/UserDaoImpl.java:31]: Make it like that:
```
    public Optional<Product> update(String name, Product newProduct) {
        Optional<Product> deletedProduct = delete(name);
        if (deletedProduct.isPresent()) {
            add(newProduct);
        }
        return deletedProduct;
    }
```

**Review by tupchiienko — CHANGES_REQUESTED** [review]

**Review by Oleksandr0404 — APPROVED** [review]

**Review by tupchiienko — APPROVED** [review]

**Pull request merged**